app/domain/service-request/functions.server.ts

```diff
@@ -139,8 +195,12 @@ export async function createServiceRequestAppData(appData: ServiceRequestAppData
   return cid;
 }
 
-async function getServiceRequestConfig(address: string, id: string) {
+async function getServiceRequestConfig(address: string, id: string, event?: TracerEvent) {
   const contract = LaborMarket__factory.connect(address, nodeProvider);
+  if (event) {
+    const data = await contract.serviceRequests(id, { blockTag: event?.block.number });
+    return ServiceRequestConfigSchema.parse(data);
```

Comment:
I think you only need to do:
```ts
const data = await contract.serviceRequests(id, { blockTag: event?.block.number });
```
and in the event where it's undefined it will just send `{}` to the function, which is fune